src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -576,17 +577,15 @@ impl AppendOnlyTopNCacheTrait for TopNCache<true> {
             self.low.is_empty(),
             "Offset is not supported yet for WITH TIES, so low cache should be empty"
         );
-
-        let elem_to_compare_with_middle = (cache_key, row);
+        let elem_to_compare_with_middle = (cache_key, row_ref);
 
         if !self.is_middle_cache_full() {
-            managed_state.insert(elem_to_compare_with_middle.1.clone());
-            self.middle.insert(
-                elem_to_compare_with_middle.0.clone(),
-                (&elem_to_compare_with_middle.1).into(),
-            );
+            let row = elem_to_compare_with_middle.1.to_owned_row();
```

Comment:
We've implemented `Into<CompactedRow>` for all `Row2`s, so there's no need to `to_owned` here.

```suggestion
            let row: CompactedRow = elem_to_compare_with_middle.1.into();
```